### PR TITLE
Add documentation for partition_setup

### PR DESCRIPTION
**Pull request by DrMullings**
Added function documentation for partition setup module

- Related ticket: https://progress.opensuse.org/issues/57752

### CODE REVIEW

File: lib/partition_setup.pm [hunk 1]
```diff
@@ -63,6 +78,10 @@ sub wipe_existing_partitions_storage_ng {
 }
 
 
+=head2 create_new_partition_table
+
+=cut
+
```
Comment (asdil12) [inline lib/partition_setup.pm]: Please remove the empty line between `=cut` and the `sub …`

File: lib/partition_setup.pm [hunk 2]
```diff
@@ -63,6 +78,10 @@ sub wipe_existing_partitions_storage_ng {
 }
 
 
+=head2 create_new_partition_table
+
```
Comment (asdil12) [inline lib/partition_setup.pm]:
```suggestion

 create_new_partition_table($table_type);
 
 [FIXME: Describe here what the function does and what the args do.]
 
```

File: lib/partition_setup.pm [hunk 3]
```diff
@@ -119,7 +138,12 @@ sub create_new_partition_table {
     }
 }
 
-# Set mount point and volume label
+=head2 mount_device
+
```
Comment (asdil12) [inline lib/partition_setup.pm]:
```suggestion

 mount_device($mount);

[FIXME: Describe here what the function does and what the args do.]

```

File: lib/partition_setup.pm [hunk 4]
```diff
@@ -119,7 +138,12 @@ sub create_new_partition_table {
...
+ Set mount point and volume label
+
+=cut
+
```
Comment (asdil12) [inline lib/partition_setup.pm]: Please remove this empty line

File: lib/partition_setup.pm [hunk 5]
```diff
@@ -139,7 +163,12 @@ sub mount_device {
     }
 }
 
-# Set size when adding or resizing partition
+=head2 set_partition_size
+
```
Comment (asdil12) [inline lib/partition_setup.pm]:
```suggestion

 set_partition_size([size => $size]);

[FIXME: Describe here what the function does and what the args do. (and that $size is in mb)]

```

File: lib/partition_setup.pm [hunk 6]
```diff
@@ -139,7 +163,12 @@ sub mount_device {
...
+ Set size when adding or resizing partition
+
+=cut
+
```
Comment (asdil12) [inline lib/partition_setup.pm]: Please remove empty line.

File: lib/partition_setup.pm (hunk 1 above)
Comment (DrMullings) [reply lib/partition_setup.pm]: According to the coding style in the parent ticket we want this line?

**Comment by DrMullings** [comment]
@SergioAtSUSE  please review

File: lib/partition_setup.pm (hunk 1 above)
Comment (asdil12) [reply lib/partition_setup.pm]: We did it without that empty line everywhere so I updated the parent ticket.

Comment (DrMullings) [reply lib/partition_setup.pm]: Adaption will come soonish

**Comment by DrMullings** [comment]
Updated

**Comment by asdil12** [comment]
The spacing seems to be fine but there are some TODOs left and also some functions are missing the function header with args in the docs.

**Comment by DrMullings** [comment]
I used TODOs instead of FIXMEs

**Comment by asdil12** [comment]
So this is still WIP?

**Comment by DrMullings** [comment]
Well if we have missing stuff I need to fix it first

**Comment by DrMullings** [comment]
I would consider it as ready as the rest should be done by the person doing the TODOs

File: lib/partition_setup.pm [hunk 7]
```diff
@@ -139,7 +170,11 @@ sub mount_device {
     }
 }
 
-# Set size when adding or resizing partition
+=head2 set_partition_size
+
+ Set size when adding or resizing partition
```
Comment (SergioAtSUSE) [inline lib/partition_setup.pm]:
```suggestion
set_partition_size([size => $size]);
  
The function can be executed when the SUT is on the yast partitioner panel `Add partition on /dev/xxx -> New Partition Size` to set the C<$args{size}> in megabytes.

Example:

set_patition_size(size => '100')
```

File: lib/partition_setup.pm [hunk 8]
```diff
@@ -40,7 +40,17 @@ our %partition_roles = qw(
   raw alt-a
 );
 
-# We got changes to the storage-ng UI in SLE 15 SP1, Leap 15.1 and TW
+=head1 PARTITION_SETUP
+
+=head2 SYNOPSIS
```
[the remark this answers is not in the document]
Comment (DrMullings) [reply lib/partition_setup.pm]: the description is in the function header where it belongs

File: lib/partition_setup.pm [hunk 9]
```diff
@@ -49,6 +59,11 @@ sub is_storage_ng_newui {
     );
 }
 
+=head2 wipe_existing_pastitions_storage_ng
+
+TODO describe what the function does and what the args do
```
[the remark this answers is not in the document]
Comment (DrMullings) [reply lib/partition_setup.pm]: We need to begin somewhere, at least we have the existance documented then

[the remark this answers is not in the document]
Comment (SergioAtSUSE) [reply lib/partition_setup.pm]: TODO is no documentation. We can do better than that :wink:

File: lib/partition_setup.pm (hunk 8 above)
[the remark this answers is not in the document]
Comment (SergioAtSUSE) [reply lib/partition_setup.pm]: I mean the module description, like in: https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/lib/power_action_utils.pm#L3

I thought you left SYNOPSIS as placeholder.

File: lib/partition_setup.pm [hunk 10]
```diff
@@ -40,7 +40,21 @@ our %partition_roles = qw(
...
+
+This module contains functions for the partitioning part of the installation
+
+=cut
+
+=head2 is_storage_ng_newui
+
+ is_storage_ng_newui();
+
+We got changes to the storage-ng UI in SLE 15 SP1, Leap 15.1 and TW
```
Comment (SergioAtSUSE) [inline lib/partition_setup.pm]:
```suggestion
Returns true if running on a scenario that expects storage-ng.
We got changes to the storage-ng UI in SLE 15 SP1, Leap 15.1 and TW
```

File: lib/partition_setup.pm [hunk 11]
```diff
@@ -157,8 +202,14 @@ sub set_partition_size {
...
 }
 
-# Method assumes that correct disk is already selected
-# Sleect Maximum size by default
+=head2 resize_partition
+
+ resize_partition();
+
+Method assumes that correct disk is already selected
+Sleect Maximum size by default
```
Comment (SergioAtSUSE) [inline lib/partition_setup.pm]:
```suggestion
Select Maximum size by default
```

File: lib/partition_setup.pm [hunk 12]
```diff
@@ -176,6 +227,13 @@ sub resize_partition {
     send_key((is_storage_ng) ? "$cmd{next}" : "$cmd{ok}");
 }
 
+=head2 addpart
+
+ addpart();
```
Comment (SergioAtSUSE) [inline lib/partition_setup.pm]:
```suggestion
 addpart(size => $size, role => $role [, format => $format] [, enable_snapshots => $enable_snapshots] [, fsid => $fsid] [, mount => $mount] [, encrypt => $encrypt]);
```

File: lib/partition_setup.pm [hunk 13]
```diff
@@ -176,6 +227,13 @@ sub resize_partition {
...
+
+TODO describe what the function does and what the args do
```
Comment (SergioAtSUSE) [inline lib/partition_setup.pm]: Please, add proper description following `assert_screen` as template: https://github.com/os-autoinst/os-autoinst/blob/8525ef05bb720c5f9a54a9a4d7fec5d9575e9406/testapi.pm#L390

File: lib/partition_setup.pm [hunk 14]
```diff
@@ -235,6 +293,12 @@ sub addpart {
     send_key(is_storage_ng() ? $cmd{next} : $cmd{finish});
 }
 
+=head2 addvg
+
+ addvg();
```
Comment (SergioAtSUSE) [inline lib/partition_setup.pm]:
```suggestion
 addvg(name => $name [, add_all_pvs => $add_all_pvs]);
```

File: lib/partition_setup.pm [hunk 15]
```diff
@@ -235,6 +293,12 @@ sub addpart {
...
+TODO describe what the function does and what the args do
```
Comment (SergioAtSUSE) [inline lib/partition_setup.pm]: Please, add proper description following `assert_screen` as template: https://github.com/os-autoinst/os-autoinst/blob/8525ef05bb720c5f9a54a9a4d7fec5d9575e9406/testapi.pm#L390

File: lib/partition_setup.pm [hunk 16]
```diff
@@ -264,6 +328,12 @@ sub addvg {
     send_key(is_storage_ng() ? $cmd{next} : $cmd{finish});
 }
 
+=head2 addlv
+
+ addlv();
```
Comment (SergioAtSUSE) [inline lib/partition_setup.pm]:
```suggestion
 addlv(vg => $vg, name => $name, role => $role [, size => $size] [, mount => $mount] [, [thinpool => $thinpool] | [thinvolume => $thinvolume]]);
```

File: lib/partition_setup.pm [hunk 17]
```diff
@@ -264,6 +328,12 @@ sub addvg {
...
+TODO describe what the function does and what the args do
```
Comment (SergioAtSUSE) [inline lib/partition_setup.pm]: Please, add proper description following `assert_screen` as template: https://github.com/os-autoinst/os-autoinst/blob/8525ef05bb720c5f9a54a9a4d7fec5d9575e9406/testapi.pm#L390

File: lib/partition_setup.pm [hunk 18]
```diff
@@ -319,6 +389,12 @@ sub addlv {
     assert_screen 'expert-partitioner';
 }
 
+=head2 addboot
+
+ addboot();
+TODO describe what the function does and what the args do
```
Comment (SergioAtSUSE) [inline lib/partition_setup.pm]: Please, add proper description following `assert_screen` as template: https://github.com/os-autoinst/os-autoinst/blob/8525ef05bb720c5f9a54a9a4d7fec5d9575e9406/testapi.pm#L390

File: lib/partition_setup.pm [hunk 19]
```diff
@@ -349,6 +425,12 @@ sub addboot {
     }
 }
 
+=head2 skip_select_first_hard_disk
+
+ skip_select_first_hard_disk();
+TODO describe what the function does and what the args do
```
Comment (SergioAtSUSE) [inline lib/partition_setup.pm]: Please, add proper description following `assert_screen` as template: https://github.com/os-autoinst/os-autoinst/blob/8525ef05bb720c5f9a54a9a4d7fec5d9575e9406/testapi.pm#L390

File: lib/partition_setup.pm [hunk 20]
```diff
@@ -358,6 +440,12 @@ sub skip_select_first_hard_disk {
     return 0;
 }
 
+=head2 select_first_hard_disk
+
+ select_first_hard_disk();
+TODO describe what the function does and what the args do
```
Comment (SergioAtSUSE) [inline lib/partition_setup.pm]: Please, add proper description following `assert_screen` as template: https://github.com/os-autoinst/os-autoinst/blob/8525ef05bb720c5f9a54a9a4d7fec5d9575e9406/testapi.pm#L390

File: lib/partition_setup.pm [hunk 21]
```diff
@@ -402,6 +495,12 @@ sub enable_encryption_guided_setup {
     installation_user_settings::await_password_check;
 }
 
+=head2 take_first_disk_storage_ng
+
+ take_first_disk_storage_ng();
+TODO describe what the function does and what the args do
```
Comment (SergioAtSUSE) [inline lib/partition_setup.pm]: Please, add proper description following `assert_screen` as template: https://github.com/os-autoinst/os-autoinst/blob/8525ef05bb720c5f9a54a9a4d7fec5d9575e9406/testapi.pm#L390

File: lib/partition_setup.pm [hunk 22]
```diff
@@ -457,6 +556,12 @@ sub take_first_disk_storage_ng {
     }
 }
 
+=head2 take_first_disk
+
+ take_first_disk();
+TODO describe what the function does and what the args do
```
Comment (SergioAtSUSE) [inline lib/partition_setup.pm]: Please, add proper description following `assert_screen` as template: https://github.com/os-autoinst/os-autoinst/blob/8525ef05bb720c5f9a54a9a4d7fec5d9575e9406/testapi.pm#L390

File: lib/partition_setup.pm [hunk 23]
```diff
@@ -457,6 +556,12 @@ sub take_first_disk_storage_ng {
     }
 }
 
+=head2 take_first_disk
+
+ take_first_disk();
```
Comment (SergioAtSUSE) [inline lib/partition_setup.pm]:
```suggestion
 take_first_disk([iscsi => $iscsi]);
```

**Comment by DrMullings** [comment]
Currently sqashing your commits and then trying to write some stuff for the missing functions

**Comment by DrMullings** [comment]
Updated.

File: lib/partition_setup.pm [hunk 24]
```diff
@@ -235,6 +295,12 @@ sub addpart {
     send_key(is_storage_ng() ? $cmd{next} : $cmd{finish});
 }
 
+=head2 addvg
+
+ addvg(name => $name [, add_all_pvs => $add_all_pvs]);
+Adds a volume group
```
Comment (SergioAtSUSE) [inline lib/partition_setup.pm]:
```suggestion
Adds an LVM volume group
```

File: lib/partition_setup.pm [hunk 25]
```diff
@@ -264,6 +330,12 @@ sub addvg {
     send_key(is_storage_ng() ? $cmd{next} : $cmd{finish});
 }
 
+=head2 addlv
+
+ addlv(vg => $vg, name => $name, role => $role [, size => $size] [, mount => $mount] [, [thinpool => $thinpool] | [thinvolume => $thinvolume]]);
+Adds a logical volume
```
Comment (SergioAtSUSE) [inline lib/partition_setup.pm]:
```suggestion
Adds an LVM logical volume
```

File: lib/partition_setup.pm [hunk 26]
```diff
@@ -402,6 +497,13 @@ sub enable_encryption_guided_setup {
     installation_user_settings::await_password_check;
 }
 
+=head2 take_first_disk_storage_ng
+
+ take_first_disk_storage_ng();
+Only works on storage ng and is being called by C<take_first_disk>
```
Comment (SergioAtSUSE) [inline lib/partition_setup.pm]:
```suggestion
Only works on storage-ng and is being called by C<take_first_disk>
```

File: lib/partition_setup.pm [hunk 27]
```diff
@@ -457,6 +559,11 @@ sub take_first_disk_storage_ng {
     }
 }
 
+=head2 take_first_disk
+
+ take_first_disk([iscsi => $iscsi]);
```
Comment (SergioAtSUSE) [inline lib/partition_setup.pm]: Missing description here.

Comment (DrMullings) [reply lib/partition_setup.pm]: I have honestly no clue what it is doing exactly, of course I can write "takes first disk" but that would add no value

**Comment by DrMullings** [comment]
Now its adapted

File: lib/partition_setup.pm [hunk 28]
```diff
@@ -235,6 +295,12 @@ sub addpart {
     send_key(is_storage_ng() ? $cmd{next} : $cmd{finish});
 }
 
+=head2 addvg
+
+ addvg(name => $name [, add_all_pvs => $add_all_pvs]);
+Adds a LVM volume group
```
Comment (SergioAtSUSE) [inline lib/partition_setup.pm]: It starts with vowel sound _/ɛl/_

```suggestion
Adds an LVM volume group
```

File: lib/partition_setup.pm [hunk 29]
```diff
@@ -264,6 +330,12 @@ sub addvg {
     send_key(is_storage_ng() ? $cmd{next} : $cmd{finish});
 }
 
+=head2 addlv
+
+ addlv(vg => $vg, name => $name, role => $role [, size => $size] [, mount => $mount] [, [thinpool => $thinpool] | [thinvolume => $thinvolume]]);
+Adds a LVM logical volume
```
Comment (SergioAtSUSE) [inline lib/partition_setup.pm]: Same here

```suggestion
Adds an LVM logical volume
```

File: lib/partition_setup.pm (hunk 28 above)
Comment (DrMullings) [reply lib/partition_setup.pm]: I learned only in front of a real vowel and Hs that are not being spoken?

Comment (SergioAtSUSE) [reply lib/partition_setup.pm]: https://dictionary.cambridge.org/grammar/british-grammar/a-an-and-the

**Comment by DrMullings** [comment]
And now we have legit grammar, please do not forget to sqash when merging

**Comment by SergioAtSUSE** [comment]
@DrMullings, I think that expecting a squash is dangerous. We are not used to do that so I predict a lot of wrong merges. Let's discuss it tomorrow in the retro if all team members agree to allow squashing

**Comment by DrMullings** [comment]
Then I will do it locally to not pollute the log

File: lib/partition_setup.pm [hunk 30]
```diff
@@ -402,6 +497,13 @@ sub enable_encryption_guided_setup {
     installation_user_settings::await_password_check;
 }
 
+=head2 take_first_disk_storage_ng
+
+ take_first_disk_storage_ng();
+Only works on storage-ng and is being called by C<take_first_disk>
+
```
Comment (asdil12) [inline lib/partition_setup.pm]: this line can be removed

File: lib/partition_setup.pm [hunk 31]
```diff
@@ -139,7 +175,18 @@ sub mount_device {
...
+=head2 set_partition_size
+
+ set_partition_size([size => $size]);
+  
+The function can be executed when the SUT is on the yast partitioner panel `Add partition on /dev/xxx -> New Partition Size` to set the C<$args{size}> in megabytes.
+
+Example:
+
+ set_patition_size(size => '100')
+
```
Comment (asdil12) [inline lib/partition_setup.pm]: this line can be removed

**Pull request merged**